src/errors/__tests__/encrypted_media_error.test.ts

```diff
@@ -0,0 +1,81 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import EncryptedMediaError from "../encrypted_media_error";
+
+describe("errors - EncryptedMediaError", () => {
+  it("should format an EncryptedMediaError when called with minimal arguments", () => {
+    const encryptedMediaError = new EncryptedMediaError("foo", null);
+    expect(encryptedMediaError).toBeInstanceOf(Error);
+    expect(encryptedMediaError.name).toBe("EncryptedMediaError");
+    expect(encryptedMediaError.type).toBe("ENCRYPTED_MEDIA_ERROR");
+    expect(encryptedMediaError.reason).toBe(null);
+    expect(encryptedMediaError.code).toBe("");
+    expect(encryptedMediaError.fatal).toBe(false);
+    expect(encryptedMediaError.message).toBe("EncryptedMediaError ()");
+  });
+
+  it("should be able to give a reason", () => {
+    const error = new Error("test");
+    const encryptedMediaError = new EncryptedMediaError("foo", error);
+    expect(encryptedMediaError).toBeInstanceOf(Error);
+    expect(encryptedMediaError.name).toBe("EncryptedMediaError");
+    expect(encryptedMediaError.type).toBe("ENCRYPTED_MEDIA_ERROR");
+    expect(encryptedMediaError.reason).toBe(error);
+    expect(encryptedMediaError.code).toBe("");
+    expect(encryptedMediaError.fatal).toBe(false);
+    expect(encryptedMediaError.message).toBe("EncryptedMediaError () test");
+  });
+
+  it("should be able to set it as fatal", () => {
+    const error = new Error("test");
+    const encryptedMediaError = new EncryptedMediaError("foo", error, true);
+    expect(encryptedMediaError).toBeInstanceOf(Error);
+    expect(encryptedMediaError.name).toBe("EncryptedMediaError");
+    expect(encryptedMediaError.type).toBe("ENCRYPTED_MEDIA_ERROR");
+    expect(encryptedMediaError.reason).toBe(error);
+    expect(encryptedMediaError.code).toBe("");
+    expect(encryptedMediaError.fatal).toBe(true);
+    expect(encryptedMediaError.message).toBe("EncryptedMediaError () test");
+  });
+
+  it("should filter in a valid error code", () => {
+    const encryptedMediaError = new EncryptedMediaError("MEDIA_ERR_NETWORK", null, true);
+    expect(encryptedMediaError).toBeInstanceOf(Error);
+    expect(encryptedMediaError.name).toBe("EncryptedMediaError");
+    expect(encryptedMediaError.type).toBe("ENCRYPTED_MEDIA_ERROR");
+    expect(encryptedMediaError.reason).toBe(null);
+    expect(encryptedMediaError.code).toBe("MEDIA_ERR_NETWORK");
+    expect(encryptedMediaError.fatal).toBe(true);
+    expect(encryptedMediaError.message)
+      .toBe("EncryptedMediaError (MEDIA_ERR_NETWORK)");
+  });
+
+  /* tslint:disable max-line-length */
+  it("should set a complete error message if both a valid code and a reason is given", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 43. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=440&issues=AWgpL74aUMkE2J58feAV&open=AWgpL74aUMkE2J58feAV)

[ISSUE_KEY]: AWgpL74aUMkE2J58feAV